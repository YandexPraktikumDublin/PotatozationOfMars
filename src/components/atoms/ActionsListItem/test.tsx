import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { ActionsListItem } from '.'

describe('<ActionsListItem />', () => {
  const name = 'Test name'

  it('should renders correct <ActionsListItem />', () => {
    const onClick = jest.fn()

    const wrapper = shallow(<ActionsListItem name={name} onClick={onClick} />)

    expect(wrapper.contains(name)).toBeTruthy()

    wrapper.find('button').simulate('click')

    expect(onClick).toHaveBeenCalledTimes(1)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has correct value from value prop', () => {
    const onClick = jest.fn()

    const value = 'Test value'

    const wrapper = shallow(
      <ActionsListItem name={name} value={value} onClick={onClick} />
    )

    expect(wrapper.contains(value)).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
