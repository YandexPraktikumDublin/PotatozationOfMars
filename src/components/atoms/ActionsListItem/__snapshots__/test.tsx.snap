// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionsListItem /> should has correct value from value prop 1`] = `
<li
  className="border-b border-primary mb-4 last:mb-0 dark:border-white"
>
  <button
    className="flex justify-between w-full text-left"
    onClick={[MockFunction]}
  >
    <span>
      Test name
    </span>
    <span>
      Test value
    </span>
  </button>
</li>
`;

exports[`<ActionsListItem /> should renders correct <ActionsListItem /> 1`] = `
<li
  className="border-b border-primary mb-4 last:mb-0 dark:border-white"
>
  <button
    className="flex justify-between w-full text-left"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  >
    <span>
      Test name
    </span>
    <span />
  </button>
</li>
`;
